include/rocksdb/filter_policy.h

```diff
@@ -90,6 +90,15 @@ class FilterPolicy : public Customizable {
   virtual ~FilterPolicy();
   static const char* Type() { return "FilterPolicy"; }
 
+  // The name used for identifying whether a filter on disk is readable
+  // by this FilterPolicy. This e.g. allows BloomFilterPolcy and
+  // RibbonFilterPolicy to have distinct Customizable names but able to
+  // read each others filters. Filter policies not based on built-in
+  // implementations may return Name(), but this is pure virtual so that
+  // wrappers around built-in policies are prompted to defer to
+  // CompatibilityName() of the wrapped policy.
+  virtual const char* CompatibilityName() const = 0;
```

Comment:
I find this comment confusing and not sure what I would do if I needed to implement this, except to just make it return Name().  Why not have the default implementation return Name() and override it for the implementations that require it?

Reply:
I understand it does the wrong thing for the builtins, but since RocksDB controls those, they are easier to test and verify they do the right thing, rather than assuming someone who writes their own having to understand what to do.

And if there is a custom one out there, this change breaks compile-time compatibility.